pkgs/development/python-modules/pytest-factoryboy/default.nix

```
@@ -0,0 +1,34 @@
+{ stdenv, fetchFromGitHub, buildPythonPackage, pythonPackages }:
+
+buildPythonPackage rec {
+  pname = "pytest-factoryboy";
+  version = "2.0.3";
+
+  src = fetchFromGitHub {
+    owner = "pytest-dev";
+    repo = "pytest-factoryboy";
+    rev = version;
+    sha256 = "0m1snyybq2k51khlydhisq300vzys897vdbsicph628iran950hn";
+  };
+
+  # The project uses tox, which we can't. So we simply run pytest manually.
+  checkInputs = with pythonPackages; [
+    pytest
+    pytestcache
+    pytestcov
+    pytestpep8
+    mock
+  ];
```

When you make these all single arguments, then you can remove ``pythonPackages``:
```suggestion
  checkInputs = [
    pytestCheckHook
    pytestcache
    pytestcov
    pytestpep8
    mock
  ];
```
I also suggest using ``pytestCheckHook`` to run pytest in the check phase.

Also, is pep8 necessary to run? That's more a backend formatting thing, and it could cause undesired Nix errors if something is slightly malformatted. I would suggest removing ``pytestpep8`` here and ``--pep8`` from ``pytestFlagsArray``.

---

I've just learned that pytest reuses `addopts = -vv -l --pep8` from the tox.ini[1].  I did not find a way to disable this behaviour. We could delete tox.ini or simply leave it as it is? What do you think?

[1] - https://docs.pytest.org/en/latest/customize.html

---

Huh, interesting. I've never seen that done before. In that case, I would either remove the addopts in ``postPatch`` (i.e. ``substituteInPlace tox.ini --replace "oldstring" ""``, or just leave it as is b/c it doesn't hurt anything.
Also, please mark other comments as resolved, it makes it easier to read this PR.